server/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -227,11 +226,14 @@ public final String getNodeName() {
      * Submits a cluster state update task; unlike {@link #submitStateUpdateTask(String, Object, ClusterStateTaskConfig,
      * ClusterStateTaskExecutor, ClusterStateTaskListener)}, submitted updates will not be batched.
      *
+     * @deprecated Unbatched cluster state update tasks are deprecated as they may cause scalability issues if submitted at high rates.
+     *             Use batched state updates via {@link #submitStateUpdateTask(String, Object, ClusterStateTaskConfig,
+     *             ClusterStateTaskExecutor, ClusterStateTaskListener)} instead.
      * @param source     the source of the cluster state update task
      * @param updateTask the full context for the cluster state update
      *                   task
-     *
      */
+    @Deprecated
```

Comment:
I wonder if we could instead let IntelliJ inline this method and remove the original method here. And then add a todo on every usage to move to batched cluster state update?

Reply:
This is used 187 times in the current codebase. Personally, I don't think I'd do that kind of noisy change (it's really annoying because we often create the task instance inline so it's not a straightforward inlining in the callers) when we could just add a meta issue or so asking to fix these?